web/plugin-manager/src/components/installNodeManager.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState } from 'react';
+import axiosServices from '../services/axios';
+import { Circles } from 'react-loader-spinner'
+
+export interface InstallProps {
+    errorHandler: (errorType: string, errorMessage: string) => void;
+    getPluginsInfo: () => void;
+}
+
+function getOperatingSystem(): string | undefined {
+    const userAgent = navigator.userAgent.toLowerCase();
+    const platform = navigator.platform.toLowerCase();
+
+    if (platform.includes("win")) {
+        return "win";
+    } else if (platform.includes("mac") || platform.includes("darwin")) {
+        if (userAgent.includes("intel")) {
+            return "macos amd64";
+        } else if (userAgent.includes("apple")) {
+            return "macos arm64";
+        }
+    } else if (platform.includes("linux")) {
+        return "linux";
+    }
+}
+
+function InstallNodeManager(p: InstallProps) {
+
+    const [isInstalling, setIsInstalling] = useState<boolean>(false);
+
+    const [os, setOs] = useState<string | undefined>(undefined);
+    useEffect(() => setOs(getOperatingSystem()), []);
+
+    async function handleInstallPlugin(_: any) {
+        if (!os) {
+            p.errorHandler("error", `Unable to detect operating system`);
+            return
+        }
+        let url = "https://github.com/massalabs/thyra-node-manager-plugin/releases/latest/download/thyra-plugin-node-manager_";
+        switch (os) {
```

Comment:
No default value ?

Reply:
good point, os could be undefined, in that case should we display a flash message or do nothing? @SlnPons  what to do when the OS could not be detected?

Reply:
The lines just before this handle this case!
```
    if (!os) {
        p.errorHandler("error", `Unable to detect operating system`);
 ```
 TBH i 'm not very proud of this stuff. and i am not even sure its working on all Os... depending on this, we may change the autodetection to a drop down or something else to let the user chose his os